processor/src/main/java/org/mapstruct/ap/internal/processor/creation/MappingResolverImpl.java

```diff
@@ -122,14 +122,13 @@ public Assignment getTargetAssignment(Method mappingMethod, String mappedElement
         ResolvingAttempt attempt = new ResolvingAttempt(
             sourceModel,
             mappingMethod,
-            mappedElement,
             dateFormat,
             numberFormat,
             sourceRHS,
             criteria
         );
 
-        return attempt.getTargetAssignment( sourceRHS.getSourceType(), targetType );
+        return attempt.getTargetAssignment( sourceRHS.getSourceTypeForMatching(), targetType );
```

Comment:
I am a bit confused here. Is this the same thing we did before? Was it wrong before, or just the source type was the element when it was an `ADDER`?